src/webapp/components/form/FormState.ts

```diff
@@ -81,50 +85,215 @@ export type FormFieldState =
     | FormDateFieldState
     | FormAvatarFieldState;
 
+export function getAllFieldsFromSections(formSections: FormSectionState[]): FormFieldState[] {
+    return formSections.reduce(
+        (acc: FormFieldState[], section: FormSectionState): FormFieldState[] => {
+            if (section.subsections) {
+                return [...acc, ...getAllFieldsFromSections(section.subsections)];
+            } else {
+                return [...acc, ...section.fields];
+            }
+        },
+        []
+    );
+}
+
+export function getStringFieldValue(id: string, allFields: FormFieldState[]): string {
+    return (
+        getFieldValueById<FormTextFieldState | FormOptionsFieldState | FormAvatarFieldState>(
+            id,
+            allFields
+        ) || ""
+    );
+}
+
+export function getDateFieldValue(id: string, allFields: FormFieldState[]): Date | null {
+    return getFieldValueById<FormDateFieldState>(id, allFields) || null;
+}
+
+export function getBooleanFieldValue(id: string, allFields: FormFieldState[]): boolean {
+    return !!getFieldValueById<FormBooleanFieldState>(id, allFields);
+}
+
+export function getMultipleOptionsFieldValue(id: string, allFields: FormFieldState[]): string[] {
+    return getFieldValueById<FormMultipleOptionsFieldState>(id, allFields) || [];
+}
+
+export function getFieldValueById<F extends FormFieldState>(
+    id: string,
+    fields: FormFieldState[]
+): F["value"] | undefined {
+    const field = fields.find(field => field.id === id);
+    if (field) {
+        return getFieldValue<F>(field);
+    }
+}
+
+export function getFieldIdFromIdsDictionary<T extends Record<string, string>>(
+    key: keyof T,
+    fieldIdsDictionary: T
+): string {
+    return fieldIdsDictionary[key] as string;
+}
+
+function getEmptyValueForField<F extends FormFieldState>(field: FormFieldState): F["value"] {
+    switch (field.type) {
+        case "text":
+            return "";
+        case "boolean":
+            return false;
+        case "select":
+            return field.multiple ? [] : "";
+        case "radio":
+            return "";
+        case "date":
+            return null;
+        case "user":
+            return undefined;
+    }
+}
+
+function getFieldValue<F extends FormFieldState>(field: FormFieldState): F["value"] {
+    return field.value;
+}
+
+// UPDATES:
+
+export function updateFormStateAndApplySideEffects(
+    currentFormState: FormState,
+    updatedField: FormFieldState
+): FormState {
+    const updatedFormState = updateFormState(currentFormState, updatedField);
+
+    return applySideEffects(updatedFormState, updatedField);
+}
+
+function applySideEffects(updatedFormState: FormState, updatedField: FormFieldState): FormState {
+    if (updatedField.notApplicableFieldId) {
```

Comment:
in the future, if we see more fields with "side-effects" then we should add a effects property to FormFieldState and have an array of fields it effects and the type of effect it has. But for now, since only NA has side effects, this looks good.